x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/downsample/DownsampleTransportFailureTests.java

```diff
@@ -0,0 +1,414 @@
+            return (MockTransportService) internalCluster().getInstance(TransportService.class, this.coordinator);
+        }
+
+        public List<MockTransportService> allMockTransportServices() {
+            return Arrays.stream(cluster.getNodeNames())
+                .map(nodeName -> (MockTransportService) internalCluster().getInstance(TransportService.class, nodeName))
+                .collect(Collectors.toList());
+        }
+
+        public String coordinatorName() {
+            assert this.coordinator != null;
+            return this.coordinator;
+        }
+
+        public String workerName() {
+            assert this.worker != null;
+            return this.worker;
+        }
+    }
+
+    private static final int DOWNSAMPLE_ACTION_TIMEOUT_MILLIS = 10_000;
+    private static final String SOURCE_INDEX_NAME = "source";
+    private static final String TARGET_INDEX_NAME = "target";
+    private long startTime;
+    private long endTime;
+
+    private final List<String> DOCUMENTS = new ArrayList<>(
+        List.of(
+            "{\"@timestamp\": \"2020-09-09T18:03:00\",\"dim1\": \"dim1\",\"dim2\": \"401\",\"gauge\": \"100\",\"counter\": \"100\"}",
+            "{\"@timestamp\": \"2020-09-09T18:04:00\",\"dim1\": \"dim1\",\"dim2\": \"402\",\"gauge\": \"101\",\"counter\": \"101\"}",
+            "{\"@timestamp\": \"2020-09-09T18:05:00\",\"dim1\": \"dim1\",\"dim2\": \"403\",\"gauge\": \"102\",\"counter\": \"102\"}",
+            "{\"@timestamp\": \"2020-09-09T18:06:00\",\"dim1\": \"dim1\",\"dim2\": \"404\",\"gauge\": \"101\",\"counter\": \"103\"}",
+            "{\"@timestamp\": \"2020-09-09T18:07:00\",\"dim1\": \"dim1\",\"dim2\": \"405\",\"gauge\": \"103\",\"counter\": \"104\"}",
+            "{\"@timestamp\": \"2020-09-09T18:08:00\",\"dim1\": \"dim1\",\"dim2\": \"406\",\"gauge\": \"110\",\"counter\": \"105\"}",
+            "{\"@timestamp\": \"2020-09-09T18:09:00\",\"dim1\": \"dim1\",\"dim2\": \"407\",\"gauge\": \"112\",\"counter\": \"106\"}",
+            "{\"@timestamp\": \"2020-09-09T18:10:00\",\"dim1\": \"dim1\",\"dim2\": \"408\",\"gauge\": \"111\",\"counter\": \"107\"}",
+            "{\"@timestamp\": \"2020-09-09T18:11:00\",\"dim1\": \"dim1\",\"dim2\": \"409\",\"gauge\": \"105\",\"counter\": \"108\"}",
+            "{\"@timestamp\": \"2020-09-09T18:12:00\",\"dim1\": \"dim1\",\"dim2\": \"410\",\"gauge\": \"106\",\"counter\": \"109\"}",
+            "{\"@timestamp\": \"2020-09-09T18:13:00\",\"dim1\": \"dim1\",\"dim2\": \"411\",\"gauge\": \"107\",\"counter\": \"110\"}",
+            "{\"@timestamp\": \"2020-09-09T18:14:00\",\"dim1\": \"dim1\",\"dim2\": \"412\",\"gauge\": \"104\",\"counter\": \"111\"}"
+        )
+    );
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return List.of(
+            LocalStateCompositeXPackPlugin.class,
+            Rollup.class,
+            AggregateMetricMapperPlugin.class,
+            DataStreamsPlugin.class,
+            IndexLifecycle.class
```

Comment:
Are all those plugins required for this test? I think that `IndexLifecycle` and `DataStreamsPlugin` as I did not see any ILM or data stream related code.